aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -140,9 +141,9 @@ private void recurseTree(
                                 childrenIds.add(cursor.getString(index));
                             } while (cursor.moveToNext());
                         }
-                    } catch (DataStoreException exception) {
+                    } catch (DataStoreException | SQLiteException exception) {
```

Comment:
Should this be a more specific SQLite foreign key constraint exception?

Reply:
This aims to ignore _any_ unexpected exception that can be thrown here, so if anything I would rather catch generic `Exception` here, but that would be code smell...
For example:
* Any SQLite specific behavior that was unexpected
  * 1000 variable/expression limit that we didn't know about would be an example of this (#1121 aims to fix this explicitly though)
* Incorrectly formatted graphQL schema that generated malformed models
  * #1146 would be an example of this. If customers started off with malformed models that happened to work out, we don't want to suddenly break their code by catching them here